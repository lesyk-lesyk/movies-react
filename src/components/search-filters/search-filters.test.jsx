import React from 'react';
import { SearchFilters } from './index';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

jest.mock('../search-mode-button/search-mode-button');

describe('SearchFilters', () => {
  it('renders', () => {
    const tree = renderer
      .create(<SearchFilters />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should have default state', () => {
    const wrapper = shallow(<SearchFilters />);

    expect(wrapper.instance().state.isTitleEnabled).toEqual(true);
    expect(wrapper.instance().state.isGenreEnabled).toEqual(false);
  });

  it('should handle Genre Click', () => {
    const wrapper = shallow(<SearchFilters />);

    expect(wrapper.instance().state.isTitleEnabled).toEqual(true);
    expect(wrapper.instance().state.isGenreEnabled).toEqual(false);

    wrapper.instance().handleGenreClick();

    expect(wrapper.instance().state.isTitleEnabled).toEqual(false);
    expect(wrapper.instance().state.isGenreEnabled).toEqual(true);
  });

  it('should handle Title Click after Genre Click', () => {
    const wrapper = shallow(<SearchFilters />);
    wrapper.instance().handleGenreClick();

    expect(wrapper.instance().state.isTitleEnabled).toEqual(false);
    expect(wrapper.instance().state.isGenreEnabled).toEqual(true);

    wrapper.instance().handleTitleClick();

    expect(wrapper.instance().state.isTitleEnabled).toEqual(true);
    expect(wrapper.instance().state.isGenreEnabled).toEqual(false);
  });
});
