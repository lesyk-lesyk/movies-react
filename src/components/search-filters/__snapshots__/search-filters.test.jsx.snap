// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchFilters renders 1`] = `
<div
  className="search-filters"
>
  <span
    className="filters-caption"
  >
    Search by
  </span>
  <SearchModeButton
    active={true}
    onClick={[Function]}
  >
    Title
  </SearchModeButton>
  <SearchModeButton
    active={false}
    onClick={[Function]}
  >
    Genre
  </SearchModeButton>
</div>
`;
